### DISCUSSION TITLE
How to add entry to Spectrum metadata

### DISCUSSION THREAD

**Discussion by vedina** [h2020charisma/ramanchada2#107] [Q&A] [answered]
How to update the metadata of a spectrum instance ? @georgievgeorgi  
This doesn't work .
```
spe.meta["sample"] = "smth"
```

**Comment by georgievgeorgi** [comment, marked as answer]
[`Spectrum.meta`](https://github.com/h2020charisma/ramanchada2/blob/2d5c3b0feaea47ba8d50ac51994ea3ffd866293b/src/ramanchada2/spectrum/spectrum.py#L179) is a getter for [`Spectrum._metadata`](https://github.com/h2020charisma/ramanchada2/blob/2d5c3b0feaea47ba8d50ac51994ea3ffd866293b/src/ramanchada2/spectrum/spectrum.py#L33) which is an instance of [`SpeMetadataModel`](https://github.com/h2020charisma/ramanchada2/blob/2d5c3b0feaea47ba8d50ac51994ea3ffd866293b/src/ramanchada2/misc/types/spectrum/metadata.py#L58C7-L58C19) which does not provide a setter. By design Spectrum for the most part should be immutable. By altering the metadata dictionary you will be violating this concept. With `spe.meta = new_meta` you may change the metadata with new dictionary which will be validated to `SpeMetadataModel`.
If for a reason you need to modify it in-place, you can do so with `spe.meta.__root__['new_key'] = 'new_val'`
You may define also a setter in `SpeMetadataModel`, which i discourage, as it makes it easy for the end user to alter the metadata and again brakes the concept of immutability of the Spectrum instances.

**Comment by vedina** [comment]
- the problem with metadata loaded from native files is that it is not standardized and is effectively useless.  
- So if we want standardized metadata , we have to change what is read from the native files.  
- besides, we need to add metadata of the sample,  provider (e.g. organisation doing the experiment) , etc, which can not be retrieved from the native files
- what I do for uploading standardized metadata is adding a separate dataset directly in the h5file (bypassing the Spectrum class ) with standardized annotation as attributes, but it is not optimal
